### PR TITLE
Fixes #17717: systemd detection fails when init has parameters

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/17717

### CODE REVIEW

**Comment by amousset** [comment]
This produces a `bash: warning: command substitution: ignored null byte in input
` when systemd has no parameters

**Comment by amousset** [comment]
Plus it does not work as `/usr/lib/systemd/systemd--switched-root--system--deserialize31` does not match `/systemd$`

**Comment by peckpeck** [comment]
PR updated with a new commit

File: rudder-agent/debian/postinst [hunk 1]
```diff
@@ -32,7 +32,7 @@ case "$1" in
 
     CFRUDDER_USE_SYSTEMD="false"
 
-    if readlink -f $(cut -f 1 -d " " < /proc/1/cmdline) | grep -qE "/systemd$" ; then CFRUDDER_USE_SYSTEMD="true"; fi
+    if readlink -f $(strings /proc/1/cmdline | head -n1 | cut -f 1 -d " ") | grep -qE "/systemd$" ; then CFRUDDER_USE_SYSTEMD="true"; fi
```
Comment (amousset) [inline rudder-agent/debian/postinst:35]: this requires to add binutils as dependency for the agent

Comment (amousset) [reply rudder-agent/debian/postinst:35]: it is not installed by default in base docker images, and probably some stripped-down VMs too.

**Comment by peckpeck** [comment]
PR updated with a new commit

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Pull request merged**